### DISCUSSION TITLE
Help needed: Need sample files for MT's unit tests

### DISCUSSION THREAD

**Discussion by FJBDev** [mytourbook/mytourbook#939] [Q&A]
I am looking for sample files in the below formats in order to complete and finalize the unit test suite in MT.
The unit tests help stabilize and reduce bug/regression risks in MT.

Note, those files will be public in Github

~~.csv 	Daum Ergo Bike~~
**.csv 	CicloTour text export** 	
.dat Fileformat from HACtronic and MyTourbook 
~~.hrm 	Polar~~
~~.pdd 	Polar~~
**.crs 	       Garmin** 	
**.xml         Suunto Quest**
~~.tour 	HAC4Linux data format~~ 	 
~~.tur 	Exportformat from HACtronic~~	
~~.xml 	Polar Personal Trainer~~
**.fit Swimming file**

**Comment by stvzldn** [comment]
The Garmin .crs is a very old Garmin Course Database format.  Garmin Training Center (v3.6.5 last one) will read .crs but will no longer output a .crs  I used crs courses with my Edge 305
Garmin Connect will only accept .gpx or .tcx courses.
RWGPS will also not output a .crs course file.  However omb's Courses plugin for ST3 will output a .crs which I have attached a sample.

 
[crs Garmin Course.zip](https://github.com/mytourbook/mytourbook/files/9630025/crs.Garmin.Course.zip)

**Reply by FJBDev** [reply]
@stvzldn 
Thanks for the file and the info.

Looking at the file, it seems to me that the schema is the same used than in the TCX files correct ?

```
<?xml version="1.0" encoding="utf-8"?>

<TrainingCenterDatabase xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.garmin.com/xmlschemas/TrainingCenterDatabase/v2 http://www.garmin.com/xmlschemas/TrainingCenterDatabasev2.xsd" xmlns="http://www.garmin.com/xmlschemas/TrainingCenterDatabase/v2">
```

**Reply by cyc1ingsir** [reply]
Hi Frederic,
it's cool, that you are going to implement unit tests!
IIRC and I'm remembering faintly only, I might have authored the HAC4Linux import in MT.
I collected a few samples accompanied with some screen shots from the past which I think may help to write the unit tests. The .tour Format is ASCII only and human readable. I included at least one sample with more than one line for the description (The `[NOTES]` section with the `Notes1=` ... `Notes5=` prefixes). 
A few of the samples contain various markers at the end of the file. Worth a test case, I assume.
Within the zip archive, I included the corresponding raw data to the tours.
Basically, that's what the device (in my case a Ciclosport CM414) stored internally and it should be the same format, HACtronic used. If needed, I do have some raw data from a HAC5 device somewhere as well.)
To compare the results, I added screenshots from the list of tours, the raw data dat files should contain. (The screenshots are from a project I started more than a decade ago, wanting to write a successor for HAC4linux (https://gitlab.com/cyc1ingsir/openhac) including HAC5 support and using a more recent Qt framework. Reverse engineering the HAC5 format was a lengthy process back then. I'm not the author of HAC4linux though.)
Anyway, I hope the files are helpful.
[HAC4Linux_samples.zip](https://github.com/mytourbook/mytourbook/files/9631223/HAC4Linux_samples.zip)

**Reply by stvzldn** [reply]
Re: .crs as produced by omb's Courses Plugin
I believe omb kludged the TCX schema to work rather than actually implementing it.
I have a very old utility to convert between Garmin .gpx, .hst and .crs (to make courses from an activity from Edge 305 days in 2006).  It was authored by SomeJoe and posted on MotionBased before MotionBased was acquired by Garmin.  Attached as zip with release notes.

[Gpx2Crs 1.0.2.1.zip](https://github.com/mytourbook/mytourbook/files/9633668/Gpx2Crs.1.0.2.1.zip)

**Reply by FJBDev** [reply]
> Hi Frederic, it's cool, that you are going to implement unit tests!

Thanks. Actually, the unit tests are not new as I initially added them almost [2 years ago now](https://github.com/mytourbook/mytourbook/pull/252). Since I've been contributing to MT, I noticed that some bugs/regressions could have easily been avoided with some basic unit tests and since them, I added them in the GitHub build to [automatically run ](https://github.com/mytourbook/mytourbook/pull/410)them as initially, you had to run them manually in Eclipse. I've also added automated [UI tests](https://github.com/mytourbook/mytourbook/issues/875) as well as they can help detect possible bugs, ensure the success of database upgrades...etc. All in all, the [code coverage ](https://sonarcloud.io/summary/overall?id=wolfgang-ch_mytourbook)in MT is getting to a good point where almost 30% of the code is covered now and I'm looking at increasing that.
![image](https://user-images.githubusercontent.com/25235665/192011722-813fb072-e9fe-41a6-80f7-6e63a6556673.png)
The bottom line, as I am sure you can understand is to make MT stable as much as possible to ensure that users are getting a better software at each release with as less bugs/regressions as possible.


>IIRC and I'm remembering faintly only, I might have authored the HAC4Linux import in MT.

Yes, it seems like you have (mentioned [here](https://github.com/mytourbook/mytourbook/blob/f668f2ea931bd05e7662a6231737defcc9560266/bundles/net.tourbook.device/src/net/tourbook/device/hac4linux/tour/HAC4LinuxDeviceReader.java#L17)). Thanks 😄 

> I collected a few samples accompanied with some screen shots from the past which I think may help to write the unit tests. The .tour Format is ASCII only and human readable. I included at least one sample with more than one line for the description (The `[NOTES]` section with the `Notes1=` ... `Notes5=` prefixes). A few of the samples contain various markers at the end of the file. Worth a test case, I assume. Within the zip archive, I included the corresponding raw data to the tours. Basically, that's what the device (in my case a Ciclosport CM414) stored internally and it should be the same format, HACtronic used. If needed, I do have some raw data from a HAC5 device somewhere as well.) To compare the results, I added screenshots from the list of tours, the raw data dat files should contain. (The screenshots are from a project I started more than a decade ago, wanting to write a successor for HAC4linux (https://gitlab.com/cyc1ingsir/openhac) including HAC5 support and using a more recent Qt framework. Reverse engineering the HAC5 format was a lengthy process back then. I'm not the author of HAC4linux though.) Anyway, I hope the files are helpful. [HAC4Linux_samples.zip](https://github.com/mytourbook/mytourbook/files/9631223/HAC4Linux_samples.zip)

Thank you for this very valuable archive, I will look at it carefully and use it for the unit tests.

**Reply by FJBDev** [reply]
> Re: .crs as produced by omb's Courses Plugin I believe omb kludged the TCX schema to work rather than actually implementing it. I have a very old utility to convert between Garmin .gpx, .hst and .crs (to make courses from an activity from Edge 305 days in 2006). It was authored by SomeJoe and posted on MotionBased before MotionBased was acquired by Garmin. Attached as zip with release notes.
> 
> [Gpx2Crs 1.0.2.1.zip](https://github.com/mytourbook/mytourbook/files/9633668/Gpx2Crs.1.0.2.1.zip)

Thanks, I will try this utility.
Not sure what you mean by `kludged`, but what I meant is that the `.crs` file you attached just looked to me like a TCX file with an extension of `crs` instead of TCX

**Reply by stvzldn** [reply]
Are you interested in iBike.csv files produced by Newton/Aeropod output by the Issac program ( https://velocomp.com/ ) used to upload to Strava? They contain environmental data (wind speed, barometric, altimeter, temperature) in addition to speed, distance, slope, gps &c.

[iBike.zip](https://github.com/mytourbook/mytourbook/files/9635735/iBike.zip)

**Reply by FJBDev** [reply]
> Are you interested in iBike.csv files produced by Newton/Aeropod output by the Issac program ( https://velocomp.com/ ) used to upload to Strava? They contain environmental data (wind speed, barometric, altimeter, temperature) in addition to speed, distance, slope, gps &c.
> 
> [iBike.zip](https://github.com/mytourbook/mytourbook/files/9635735/iBike.zip)

I would but it looks like it's not supported or there is an issue with the file ?

![image](https://user-images.githubusercontent.com/25235665/192023662-03081be8-0ce3-48ad-a51e-e14c54e341ec.png)

**Reply by stvzldn** [reply]
Agreed, not supported by MT

**Reply by FJBDev** [reply]
@cyc1ingsir 
While I was able to import all the `.dat` files, it seems that all the `.tour` files don't import in MT. Are you seeing the same behavior ?

![image](https://user-images.githubusercontent.com/25235665/192106925-cde8e56f-7f02-40ba-9a68-7ea51046c84a.png)

**Reply by ovahead** [reply]
Hi Frederic,

here is a CicloTour DAT and TXT and a HACtronic TUR File.
If you really looking for a CVS file let me now if it should be a MS-DOS or an Excel CSV both from Excel 2003.
CicloTour can transfer the data to Excel...

BTW
It seems there is still a little issue with CRP Files, In case the pause time is greater as the movement time (especially if you forgot to end the excersize then the device stopped at 12 hours or so automatic) The times were not shown right in MTB. Please have a look at the picture.

If I delete the "pause" at the end point by point (sometimes more than 3500 points or so) with CicloTour, the data from an imported CRP looks okay.

[DAT TXT TUR.zip](https://github.com/mytourbook/mytourbook/files/9639044/DAT.TXT.TUR.zip)

![Times](https://user-images.githubusercontent.com/112829463/192106777-8f572373-f29c-40cd-b768-a67c14a1f9d1.jpg)

The CRP File with a lot of "pause" at the end.
[280412crp.txt](https://github.com/mytourbook/mytourbook/files/9639063/280412crp.txt)

**Reply by FJBDev** [reply]
@ovahead 
Thanks for the Ciclotour DAT and TXT and a HACtronic TUR File, I will look at it.

>BTW
It seems there is still a little issue with CRV Files, In case the pause time is greater as the movement time (especially if you forgot to end the excersize then the device stopped at 12 hours or so automatic) The times were not shown right in MTB. Please have a look at the picture.

Can you please mention that in the [original issue ](https://github.com/mytourbook/mytourbook/discussions/924)so that we don't mix 2 different topics here ?
Thanks!

**Reply by FJBDev** [reply]
> Hi Frederic,
> 
> here is a Ciclotour DAT and TXT and a HACtronic TUR File. If you really looking for a CVS file let me now if it should be a MS-DOS or an Excel CSV both from Excel 2003. CicloTour can transfer the data to Excel...

@ovahead  While I was able to successfully import the `.tur` and `.dat` files, it seems that the `.txt` file fails to import, are you seeing the same thing ?

![image](https://user-images.githubusercontent.com/25235665/192107439-0f40fa5b-72d4-4f7b-86af-21226c7c3a50.png)

**Reply by ovahead** [reply]
A snapshot from the same tour but from Ciclotour.
![CicloTour-Tourdata](https://user-images.githubusercontent.com/112829463/192108084-79c05cfd-f868-47c5-a928-2c7d7e1e7db9.jpg)

**Reply by ovahead** [reply]
> @ovahead While I was able to successfully import the `.tur` and `.dat` files, it seems that the `.txt` file fails to import, are you seeing the same thing ?

Yes I see the same

**Reply by FJBDev** [reply]
> @cyc1ingsir While I was able to import all the `.dat` files, it seems that all the `.tour` files don't import in MT. Are you seeing the same behavior ?
> 
> ![image](https://user-images.githubusercontent.com/25235665/192106925-cde8e56f-7f02-40ba-9a68-7ea51046c84a.png)

Actually, when debugging, this is what MT is showing as the internal error:
```
TourLog] 14:58:48 [Importing tours - 57     ] 24    INFO        10.281 s ?16 tours were imported from the file /home/frederic/Downloads/DAT.TXT.TUR/Connect7.dat
[TourLog] 14:58:48 [main - 1                 ] 25             Imported in 10.708 s
[TourLog] 14:59:00 [main - 1                 ] 26             Importing tours...
java.time.DateTimeException: Invalid value for MonthOfYear (valid values 1 - 12): 0
	at java.base/java.time.temporal.ValueRange.checkValidValue(ValueRange.java:311)
	at java.base/java.time.temporal.ChronoField.checkValidValue(ChronoField.java:717)
	at java.base/java.time.LocalDate.of(LocalDate.java:269)
	at java.base/java.time.LocalDateTime.of(LocalDateTime.java:361)
	at java.base/java.time.ZonedDateTime.of(ZonedDateTime.java:339)
	at net.tourbook.data.TourData.setTourStartTime(TourData.java:11658)
	at net.tourbook.device.hac4linux.tour.HAC4LinuxDeviceReader.processDeviceData(HAC4LinuxDeviceReader.java:380)
	at net.tourbook.importdata.RawDataManager.importTours_FromOneFile_10(RawDataManager.java:2612)
	at net.tourbook.importdata.RawDataManager.lambda$8(RawDataManager.java:2454)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:57)
	at net.tourbook.importdata.RawDataManager.importTours_FromOneFile(RawDataManager.java:2409)
	at net.tourbook.importdata.RawDataManager.importTours_FromMultipleFiles_30_OneFile(RawDataManager.java:2273)
	at net.tourbook.importdata.RawDataManager.lambda$6(RawDataManager.java:2242)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

!ENTRY net.tourbook.common 4 0 2022-09-24 14:59:09.010
!MESSAGE incorrect header check
!STACK 0
java.util.zip.ZipException: incorrect header check
	at java.base/java.util.zip.InflaterInputStream.read(InflaterInputStream.java:165)
	at java.base/java.io.FilterInputStream.read(FilterInputStream.java:107)
	at net.tourbook.device.crp.ZLibCompressionHelper.decompress(ZLibCompressionHelper.java:41)
	at net.tourbook.device.crp.CRPDataReader.getRawFilePath(CRPDataReader.java:144)
	at net.tourbook.device.crp.CRPDataReader.validateRawData(CRPDataReader.java:489)
	at net.tourbook.importdata.RawDataManager.importTours_FromOneFile_10(RawDataManager.java:2580)
	at net.tourbook.importdata.RawDataManager.lambda$8(RawDataManager.java:2496)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:57)
	at net.tourbook.importdata.RawDataManager.importTours_FromOneFile(RawDataManager.java:2409)
	at net.tourbook.importdata.RawDataManager.importTours_FromMultipleFiles_30_OneFile(RawDataManager.java:2273)
	at net.tourbook.importdata.RawDataManager.lambda$6(RawDataManager.java:2242)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Reply by cyc1ingsir** [reply]
> @cyc1ingsir While I was able to import all the `.dat` files, it seems that all the `.tour` files don't import in MT. Are you seeing the same behavior ?
```
10:12:19          Importing tours...
10:12:20 ERROR       /home/cycling/Documents/Radtouren/samples/hac4linux/20060426_Malle_KuestenKlassik_136km.tour
10:12:21          Imported in 1.315 s
```
Yep, same behavior for me. :-(
I have imported the files successfully in the past, though:
![Kuestenklassik](https://user-images.githubusercontent.com/2297037/192134517-e2594be7-d47d-4958-9ef7-616bcd24c749.png)

I wonder, if this _might_ be the root cause.
https://github.com/mytourbook/mytourbook/blob/f668f2ea931bd05e7662a6231737defcc9560266/bundles/net.tourbook.device/src/net/tourbook/device/hac4linux/tour/HAC4LinuxDeviceReader.java#L91-L100
Not having the code cloned locally, I wasn't able to find where (and how) `checkStartSequence` is called from when importing tours. How many bytes will be checked? The first line within the files have more bytes than actually being checked.
The coded referenced in in the comment seems to maybe have been moved.
Have you any exceptions?

**Reply by FJBDev** [reply]
> Yep, same behavior for me. :-( I have imported the files successfully in the past, though

This is exactly why I believe in unit tests! It's definitely not normal that you can import a file with a specific version and not import it anymore with another version. If I can avoid those headaches.....
With that being said, do you remember which version was used to import this successfully ?

> I wonder, if this _might_ be the root cause.
> 
> https://github.com/mytourbook/mytourbook/blob/f668f2ea931bd05e7662a6231737defcc9560266/bundles/net.tourbook.device/src/net/tourbook/device/hac4linux/tour/HAC4LinuxDeviceReader.java#L91-L100
> 
> Not having the code cloned locally, I wasn't able to find where (and how) `checkStartSequence` is called from when importing tours. How many bytes will be checked? The first line within the files have more bytes than actually being checked.
> The coded referenced in in the comment seems to maybe have been moved.
> Have you any exceptions?

I will investigate it and let you know

**Reply by cyc1ingsir** [reply]
> This is exactly why I believe in unit tests!

I know, I know. Regressions are just not funny. 

> With that being said, do you remember which version was used to import this successfully ?

Unfortunately, not at all.
My guess is, it was shortly after I provided this code:
https://github.com/mytourbook/mytourbook/commit/391dad19c800fc1aa851607b1ac6112c5cace9c6
I must have switched to MT back then, wanting to import my recorded tours from the previous years. The most recent HAC4Linux tour file dates back to ~December 2009~ March 2011. Not having any from 2010.

I vaguely remember trying to import an old tour file a few (3-4) years ago and discovering it not longer working. Assuming at the time, that the support for this file type having been removed. 
:thinking: But I may mix that up with trying to find the setting page for MT to work behind an authenticating corporate proxy which was the other thing I had contributed to MT in the past. However, the latter is unlikely, since it's still present: https://github.com/mytourbook/mytourbook/blob/73fb22785117e2935295469f0d93482f58465f83/bundles/net.tourbook/src/net/tourbook/proxy/PrefPageProxy.java :thinking: 

Glancing over the main import code, I can't see, why it should fail. I could imagine, it's not recognized as a supported file type. But this is wildly guessing only....

**Reply by FJBDev** [reply]
> Glancing over the main import code, I can't see, why it should fail. I could imagine, it's not recognized as a supported file type. But this is wildly guessing only....

I am still looking into it but it seems that, for some reason, the header check fails
At the moment, I am looking to find when it started failing (aka, which MT version introduced a regression)

`java.util.zip.ZipException: incorrect header check`

**Reply by cyc1ingsir** [reply]
Hi, it's almost worth an own issue, I think...
I might have found the root cause.
The file is read line by line.
https://github.com/mytourbook/mytourbook/blob/f668f2ea931bd05e7662a6231737defcc9560266/bundles/net.tourbook.device/src/net/tourbook/device/hac4linux/tour/HAC4LinuxDeviceReader.java#L329
Inside the file there are these lines:
```
[INFORMATION]
Number=13
Date=01.05.2009
Time=09:01:00.00
```
`field[0]` is parsed for every line within the Info section
https://github.com/mytourbook/mytourbook/blob/f668f2ea931bd05e7662a6231737defcc9560266/bundles/net.tourbook.device/src/net/tourbook/device/hac4linux/tour/HAC4LinuxDeviceReader.java#L352

the method setTourStartTime is called for every parsed line within the `[INFORMATION]` section.
However, start year and month etc is reset to 0 on every parsed line:
https://github.com/mytourbook/mytourbook/blob/f668f2ea931bd05e7662a6231737defcc9560266/bundles/net.tourbook.device/src/net/tourbook/device/hac4linux/tour/HAC4LinuxDeviceReader.java#L364-L368

This will cause the exception:
```
java.time.DateTimeException: Invalid value for MonthOfYear (valid values 1 - 12): 0
	at java.base/java.time.temporal.ValueRange.checkValidValue(ValueRange.java:319)
	at java.base/java.time.temporal.ChronoField.checkValidValue(ChronoField.java:718)
	at java.base/java.time.LocalDate.of(LocalDate.java:271)
	at java.base/java.time.LocalDateTime.of(LocalDateTime.java:363)
	at java.base/java.time.ZonedDateTime.of(ZonedDateTime.java:341)
	at net.tourbook.data.TourData.setTourStartTime(TourData.java:11658)
```
I haven't analyzed it in detail, but I guess, it's this commit from July 2012, which broke it:
https://github.com/mytourbook/mytourbook/commit/126bb8b14f8c0c5e09688ef1d05f564153bbee65#diff-8c94d2eedea246587279314340981860d0665c00bf396fc275447e6ffd656d14

**Reply by FJBDev** [reply]
@cyc1ingsir 
Actually, I spent several hours looking into this issue and I have made a fix for it for MT's next version.
Here is what your provided files look like now in MT [after my fix](https://github.com/FJBDev/mytourbook/pull/96/commits/ab06da6435891a577e694f792b839f385af05fd3)

![image](https://user-images.githubusercontent.com/25235665/193635556-250dfa56-0966-4b6a-b60d-c1381732f61c.png)

After a lot of excavating, I think you are right about the code that broke it.
I found that those `.tour` files stopped being able to be imported in 12.12.0


![image](https://user-images.githubusercontent.com/25235665/193635940-297eb3fd-e3b4-4404-be42-2409494bd5ff.png)

![image](https://user-images.githubusercontent.com/25235665/193635985-1b1f70f7-f2ee-40d6-b9d4-2bfe4f8285d9.png)

**Reply by FJBDev** [reply]
@cyc1ingsir

I have added unit tests with one of your provided file so now, we should know when the import has a regression in the future 😉 
Thanks for providing sample files !

**Reply by cyc1ingsir** [reply]
@FJBDev 
Thanks! That's good news!
You've beaten me to it.
This was my approach of fixing the issue:
https://github.com/mytourbook/mytourbook/compare/main...cyc1ingsir:mytourbook:fix/hac4linux_tour_import
Even assuming, HAC4Linux (https://hac4linux.sourceforge.net/scripts/desktop.php) will have always written Date prior to Time to the file, I wanted to make the import not dependent on the correct order.
However, I'm glad, the import is working again. In case I come across some more of the older files. Thank you for looking into the issue.

Also, I noticed, the method `checkStartSequence` seems to be never been called and it's implementation could be removed.

**Reply by FJBDev** [reply]
> @FJBDev Thanks! That's good news! You've beaten me to it. This was my approach of fixing the issue:  Thank you for looking into the issue.

You're welcome. Are you interested in testing a beta version before the official release ?
> 
> Also, I noticed, the method `checkStartSequence` seems to be never been called and it's implementation could be removed.

That I'm not sure. Maybe @wolfgang-ch would know more if it's actually called or if it should be removed but I see it's being used [here](https://github.com/FJBDev/mytourbook/blob/b4669d08276f46afcefd5b7221640d5d341c38e2/bundles/net.tourbook.device/src/net/tourbook/device/PortThread.java#L50) so i would leave it.

**Reply by FJBDev** [reply]
> Hi Frederic,
> 
> here is a CicloTour DAT and TXT and a HACtronic TUR File. If you really looking for a CVS file let me now if it should be a MS-DOS or an Excel CSV both from Excel 2003. CicloTour can transfer the data to Excel...
>
> [DAT TXT TUR.zip](https://github.com/mytourbook/mytourbook/files/9639044/DAT.TXT.TUR.zip)

@ovahead 
Thanks for the files again.
I am seeing an import issue with the following file in your provided archive `130703.txt`

When I look at the [CicloTour import code in MT](https://github.com/mytourbook/mytourbook/blob/f8b1cb7b6eed3cc7a736e31cf7fbe1497bba79e5/bundles/net.tourbook.device/src/net/tourbook/device/ciclotour/text/CiclotourTextDataReader.java#L42), it seems that the below header is looked for 
`Zeit:	Strecke:	Höhe:	Geschw:	Puls:	Temperatur:	Prozent:	Cadence:`

but your file has the below header, which is different and explains why the file doesn't import.
`Fahrzeit:	Uhrzeit:	Strecke:	Höhe:	Geschw:	Puls:	Temperatur:	Prozent:	Cadence:`

Do you know if it's a new header ? an old one ?

**Reply by wolfgang-ch** [reply]
> > Also, I noticed, the method `checkStartSequence` seems to be never been called and it's implementation could be removed.
> 
> That I'm not sure. Maybe @wolfgang-ch would know more if it's actually called or if it should be removed but I see it's being used [here](https://github.com/FJBDev/mytourbook/blob/b4669d08276f46afcefd5b7221640d5d341c38e2/bundles/net.tourbook.device/src/net/tourbook/device/PortThread.java#L50) so i would leave it.

`checkStartSequence` is currently used in one place when tours are transfered with the COM port, for any other tour types it is not necessary

![image](https://user-images.githubusercontent.com/1283445/193737619-337e7ca1-c196-4833-9ebf-4698b353af65.png)

![image](https://user-images.githubusercontent.com/1283445/193737948-8c4fcd8f-190c-4042-a768-5cd71c7f1911.png)

**Comment by wolfgang-ch** [comment]
[Daum Ergometer + tur HACtronic.zip](https://github.com/mytourbook/mytourbook/files/9643188/Daum.Ergometer.%2B.tur.HACtronic.zip)

I also have a lot of HAC4/5 files but each file contains 20+ tours (it's like a ring buffer) which I don't want to be public

**Comment by FJBDev** [comment]
Added the need of FIT files containing swimming activities (I swim very rarely 😄 )

**Reply by wolfgang-ch** [reply]
Some FIT swimm files from Doriano

[swim-data.zip](https://github.com/mytourbook/mytourbook/files/11330652/swim-data.zip)

**Reply by FJBDev** [reply]
> Some FIT swimm files from Doriano
> 
> [swim-data.zip](https://github.com/mytourbook/mytourbook/files/11330652/swim-data.zip)

Do you know if Doriano gave his permission for those files to be "public" and therefore be used in the unit tests ?

**Reply by wolfgang-ch** [reply]
Sure, otherwise I've not published it

![image](https://user-images.githubusercontent.com/1283445/234778517-9a25ceca-7831-455d-9b7b-6eb1117ff6a4.png)

**Reply by FJBDev** [reply]
> Sure, otherwise I've not published it.

Superb, thank you Doriano!

**Reply by maclura** [reply]
You are welcome, FJBDev
